src/main/java/com/google/api/codegen/metadatagen/py/PythonPackageCopier.java

```diff
@@ -0,0 +1,120 @@
+/* Copyright 2016 Google Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.api.codegen.metadatagen.py;
+
+import com.google.api.codegen.metadatagen.PackageCopier;
+import com.google.api.codegen.metadatagen.PackageMetadataGenerator;
+import com.google.api.tools.framework.snippet.Doc;
+import com.google.api.tools.framework.tools.ToolOptions;
+import com.google.common.collect.ImmutableMap;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.FileVisitResult;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.SimpleFileVisitor;
+import java.nio.file.attribute.BasicFileAttributes;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * A PackageCopier specialized to calculate Python namespace packages and generate __init__.py
+ * files.
+ */
+public class PythonPackageCopier implements PackageCopier {
+
+  /** Copies gRPC source while computing namespace packages and generating __init__.py. */
+  private class PythonPackageFileVisitor extends SimpleFileVisitor<Path> {
+    ImmutableMap.Builder<String, Doc> docBuilder = new ImmutableMap.Builder<String, Doc>();
+    List<String> pythonNamespacePackages = new ArrayList<>();
+    Path inputPath;
+    String apiVersion;
+
+    /**
+     * Constructor.
+     *
+     * @param inputPath The path to the (unprocessed) gRPC source code.
+     * @param apiVersion The major version of the API.
+     */
+    public PythonPackageFileVisitor(Path inputPath, String apiVersion) {
+      this.inputPath = inputPath;
+      this.apiVersion = apiVersion;
+    }
+
+    @Override
+    public FileVisitResult visitFile(Path file, BasicFileAttributes attr) throws IOException {
+      String outFile = inputPath.relativize(file).toString();
+      docBuilder.put(
+          outFile, Doc.text(new String(Files.readAllBytes(file), StandardCharsets.UTF_8)));
```

Comment:
I'm concerned about the fact that it looks like this loads the entire contents of all the files in the input directory tree into memory and the entirety of this content is passed onto the next step.